modules/features/player/src/main/java/au/com/shiftyjelly/pocketcasts/player/viewmodel/PlayerViewModel.kt

```diff
@@ -201,7 +203,16 @@ class PlayerViewModel @Inject constructor(
         if (list.isEmpty()) {
             ShelfItems.itemsList
         } else {
-            list.mapNotNull { id ->
+            val updatedList = when {
+                list.contains(ShelfItem.Bookmark.id) -> list
+                FeatureFlag.isEnabled(Feature.BOOKMARKS_ENABLED) -> {
+                    list.toMutableList().apply {
+                        add(list.size - 1, ShelfItem.Bookmark.id)
+                    }
+                }
+                else -> list
+            }
+            updatedList.mapNotNull { id ->
```

Comment:
This was a bit hard to read, how about this comment?
```suggestion
            // Add missing bookmark item if the feature flag is enabled
            val updatedList = when {
                list.contains(ShelfItem.Bookmark.id) -> list
                FeatureFlag.isEnabled(Feature.BOOKMARKS_ENABLED) -> {
                    list.toMutableList().apply {
                        add(list.size - 1, ShelfItem.Bookmark.id)
                    }
                }
                else -> list
            }
            updatedList.mapNotNull { id ->
```